readthedocs/core/middleware.py

```diff
@@ -61,7 +61,7 @@ def process_request(self, request):
         # Serve CNAMEs
         if (
             public_domain not in host and
-            settings.PRODUCTION_DOMAIN not in host and
+            settings.PRODUCTION_DOMAIN not in (host, full_host) and
```

Comment:
This isn't the same as the previous logic. This is checking that production_domain is exactly the same as host or full_host. Or that is what this change is for?

Reply:
Yes. That's the intended change and it's because in development the PRODUCTION_DOMAIN contains the port which is not in `host` but it's on `full_host`.